lib/src/model/puzzle/puzzle_providers.dart

```diff
@@ -34,11 +35,16 @@ Future<PuzzleStreakResponse> streak(StreakRef ref) {
   return Result.release(repo.streak());
 }
 
-// TODO when history database is available should first try to fetch from there
 @Riverpod(keepAlive: true)
-Future<Puzzle> puzzle(PuzzleRef ref, PuzzleId id) {
```

Comment:
the return type should stay `Future<Puzzle?>` imo; we're not certain the puzzle exist in the db;